### DISCUSSION TITLE
encoding error on debian 12.5

### DISCUSSION THREAD

**Discussion by awaum** [Kidel/Kathara#273] [Q&A] [answered]
Salve,

when I run `kathara settings` or `kathara linfo` I get an error:

```
Exception in thread Thread-8 (_wrap_start):
Traceback (most recent call last):
  File "/usr/lib/kathara/threading.py", line 1045, in _bootstrap_inner
  File "/usr/lib/kathara/threading.py", line 982, in run
  File "/usr/lib/kathara/Kathara/trdparty/consolemenu/console_menu.py", line 166, in _wrap_start
  File "/usr/lib/kathara/Kathara/trdparty/consolemenu/console_menu.py", line 222, in _main_loop
  File "/usr/lib/kathara/Kathara/trdparty/consolemenu/console_menu.py", line 229, in draw
  File "/usr/lib/kathara/Kathara/trdparty/consolemenu/screen.py", line 72, in printf
UnicodeEncodeError: 'ascii' codec can't encode characters in position 3-77: ordinal not in range(128)
```

System details:
```
*	Current Manager is: Docker (Kathara)
*	Manager version is: 25.0.3
*	Python version is: 3.11.7 (main, Dec  8 2023, 18:56:58) [GCC 11.4.0]
*	Kathara version is: 3.7.1
*	Operating System version is: Linux-6.1.0-18-arm64-aarch64
*	Trying to run container with `kathara/base` image...
Deploying devices...|###########################################################################################| 1/1
*	Container run successfully.
Deleting devices...|############################################################################################| 1/1
```

I found [this hint](https://stackoverflow.com/questions/9942594/unicodeencodeerror-ascii-codec-cant-encode-character-u-xa0-in-position-20), but don't know how to fix.

Can somebody help me?

Thank you!

**Comment by tcaiazzi** [comment]
HI @awaum, 

Sorry for the delay, and thanks for opening the issue. 

We never experimented this error. And, unfortunately, I'm unable to reproduce it. 

However, I don't think it is related to Debian, since I mainly develop on Debian/Ubuntu distros. 

Maybe, is something related to the encoding used in your terminal. 

Can you please provide me some additional context? On which terminal are you trying to execute the commands? What is the enconding of your terminal?

Thanks!

**Reply by awaum** [reply, marked as answer]
Ciao @tcaiazzi,

thank you for your hint with the encoding.

Well, all I did was to install [this image](https://raspi.debian.net/tested/20231109_raspi_4_bookworm.img.xz) from [Tested images](https://raspi.debian.net/tested-images/) and installed Kathara, following the instructions for "Debian 12" on your [installation guide](https://github.com/KatharaFramework/Kathara/wiki/Linux), and installed Docker of course.

However, I was able to find [this discussion here](https://stackoverflow.com/a/57448568):
`export PYTHONIOENCODING=utf8`
solved my problem :-D

Thank you!

**Reply by awaum** [reply]
just an addition:
The problem returned, though that variable was set.
This time I was able to fix it by defining the system locale (which seems to be undefined by default).
`sudo localectl set-locale LANG=en_US.UTF-8`
